components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -182,48 +180,43 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         }
 
         if(StringUtils.isBlank(openIdUrl)){
-            openIdUrl = serverUrl + "openid";
+            openIdUrl = IdentityUtil.getServerURL("openid", true);
         }
         if(StringUtils.isBlank(samlSSOUrl)){
-            samlSSOUrl = serverUrl + "samlsso";
+            samlSSOUrl = IdentityUtil.getServerURL("samlsso", true);
         }
         if(StringUtils.isBlank(samlLogoutUrl)){
-            samlLogoutUrl = serverUrl + "samlsso";
+            samlLogoutUrl = IdentityUtil.getServerURL("samlsso", true);
         }
         if(StringUtils.isBlank(oauth1RequestTokenUrl)){
-            oauth1RequestTokenUrl = serverUrl + "oauth/request-token";
+            oauth1RequestTokenUrl = IdentityUtil.getServerURL("oauth/request-token", true);
         }
         if(StringUtils.isBlank(oauth1AuthorizeUrl)){
-            oauth1AuthorizeUrl = serverUrl + "oauth/authorize-url";
+            oauth1AuthorizeUrl = IdentityUtil.getServerURL("oauth/authorize-url", true);
         }
         if(StringUtils.isBlank(oauth1AccessTokenUrl)){
-            oauth1AccessTokenUrl = serverUrl + "oauth/access-token";
+            oauth1AccessTokenUrl = IdentityUtil.getServerURL("oauth/access-token", true);
```

Comment:
can we keep these inline string values as constants [eg:"oauth2"].I noticed some of those are using from another code blocks as well.